### DISCUSSION TITLE
No permission to retrieve LAPS Password or no LAPS Password set!

### DISCUSSION THREAD

**Discussion by c-cuthbert** [Seji64/LAPS-WebUI#26] [Q&A]
I am receiving this message when attempting to retrieve the LAPS password for computers. I have confirmed the computers have LAPS password set and I'm using a Domain Admin account which definitely has permission.

I'm not seeing anything helpful in logging to narrow down whats causing this. Would really appreciate any ideas!

**Comment by Seji64** [comment]
Does the DN or the OU where the computer is located has any special characters ?

**Comment by c-cuthbert** [comment]
Hi there,

No, I've tried a couple of OUs, all behave same way.

Regards,

Christian

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Seji64** [comment]
Hm okay. Do you mind to shtare more information about your environment?

- Docker or native?
- OS
- Is LAPS-WebUI and the domain controller located in different subnets?

**Comment by c-cuthbert** [comment]
Hi there,

Running the Docker LAPS WebUI
Testing from Windows 11 OS
Different subnet to DC, although I have tested on same subnet, problem is still present.

Regards,

Christian

**Comment by dmurdoch7** [comment]
Hi there,

Just wondering if you received any update on this. I am getting this error as well and have delegated control for the user that queries the AD. Are there other permissions that are required for the user to work. I have installed LAPS Web UI on IIS and not locally.

Regards,
Darren

**Comment by Seji64** [comment]
Since Laps-WebUi brings its own web server, it is honestly not clear to me how you did it that it runs in the IIS. Any way: does the IIS run under a corresponding user who also has these delegated rights in the AD? Is Python with all required Modules installed?

**Reply by dmurdoch7** [reply]
To get it onto IIS I just opened the solution in Visual Studio 2022 and then published it to a folder. Gave it a hostname (https) and installed the python 3.13.x and I can log in and browse the computer list.

I stepped through the code and it looks like the LDAPService.cs file is not pulling all the attributes from AD (its only getting this on the line            

 LdapEntry? ldapSearchResult = (await ldapConnection.SearchAsync(defaultNamingContext, ldapFilter)).SingleOrDefault();

from the GetAdComputerAsync method but version 2 is looking for                 

string fieldName = (domain.Laps.EncryptionDisabled ? "msLAPS-Password" : "msLAPS-EncryptedPassword");


objectClass
cn
userCertificate
distinguishedName
instanceType
whenCreated
whenChanged
displayName
uSNCreated
uSNChanged
name
objectGUID
userAccountControl
badPwdCount
codePage
countryCode
badPasswordTime
lastLogoff
lastLogon
localPolicyFlags
pwdLastSet
primaryGroupID
objectSid
accountExpires
logonCount
sAMAccountName
sAMAccountType
operatingSystem
operatingSystemVersion
dNSHostName
servicePrincipalName
objectCategory
isCriticalSystemObject
dSCorePropagationData
lastLogonTimestamp
msDS-SupportedEncryptionTypes
msDS-KeyCredentialLink
msLAPS-PasswordExpirationTime

So not sure what to do there?

Regards,
Darren

**Reply by Seji64** [reply]
Check with powershell or ADUC if there is really a Laps Password present

**Comment by dmurdoch7** [comment]
@Seji64 Hey there, just wondering if you had any thoughts on my previous post?  Regards, Darren

**Comment by raxmans** [comment]
Hi,
I got your tool working, nice job.
Would it be possible, to get it working on IIS?
Or how can I use it on https?
Thanks a lot

**Reply by Seji64** [reply]
Great you got it Running. Never deployed it myself to iis. For https I suggest a reverse proxy Like haproxy,nginx or caddy.
